### PR TITLE
Escape cloud-config

### LINKED ISSUE #2434: Controller-Manager crashloops when using vsphere and a user with `ADDOMAIN\USERNAME` notation
**User Story**

Shaw reported issues of the controller-manager crashlooping when using the active directory notation of `ADDOMAIN\USERNAME`, most likely due to incorrect/no escaping in the cloud provider config

**Acceptance criteria**

* The `ADDOMAIN\USERNAME` can be used for creating a new cluster without any component crashlooping or having authentication issues

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2434

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note bugfix
cloud-config files are now properly escaped
```

### CODE REVIEW

File: api/Gopkg.toml [hunk 1]
```diff
@@ -120,7 +120,7 @@ required = [
 
 [[override]]
   name = "github.com/kubermatic/machine-controller"
-  version = "v0.10.5"
+  revision = "8fea08fd25edf2bc87511e8cfd4374bf648b5de6"
```
Comment (alvaroaleman) [inline api/Gopkg.toml]: Once this PR and kubermatic-machine/controller#423 are approved, I'll cut a machine-controller release and update here

**Comment by alvaroaleman** [comment]
@mrIncompetent PR is IMHO done, PTAL

**Comment by mrIncompetent** [comment]
looks great :)

**Pull request merged**